### PR TITLE
change dropout lowering in symbolic_script

### PR DESCRIPTION
**Pull request by ngimel**
Dropout is now eligible for fusion, and generated fused kernels are just as fast as dropout in ATen. Change its lowering in symbolic script so that it can actually be fused. Still special-cased for cuda, because without fusion this lowering is less efficient than current (bernoulli_ * input). Testing is covered by the test case that @ailzhang added (test_dropout_cuda).

### CODE REVIEW

**Review by apaszke — APPROVED** [review]

**Review by apaszke — DISMISSED** [review]
Actually wait. Why do we have backend-specific paths if you're no longer using `_fused_dropout`? We should just unify them

**Comment by ngimel** [comment]
cuda path is rand_like + comparison to get mask + type cast for mask + multiply, whereas cpu is bernoulli_ + multiply. For cuda all those operations don't matter because they are fused, for cpu, with its disabled fusions, path currently used for cuda would be much slower than bernoulli_ and multiply. If you think slowing down cpu is acceptable, I'll happily unify the paths. Unifying to bernoulli_ + multiply does not look good for fuser because 1) it requires at::empty and in-place bernoulli which fuser is not too happy about, 2) it outputs mask as float rather than as byte, which is also less efficient.

**Comment by apaszke** [comment]
Ok, that's fair. But please, leave a comment in the source that summarizes what you just said here!

File: torch/csrc/jit/symbolic_script.cpp [hunk 1]
```diff
@@ -708,7 +705,8 @@ const std::vector<std::string> functions = {
             # CUDA has a fused dropout implementation
```
Comment (fmassa) [inline torch/csrc/jit/symbolic_script.cpp]: This comment doesn't seem relevant anymore?

**Comment by ngimel** [comment]
I've disabled rocm and windows test, because we can expect bitwise accuracy only if
1) fuser is used (not true for Windows)
2) _fused_dropout is used in eager mode (not true for rocm).

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**